src/Credentials/ImpersonatedServiceAccountCredentials.php

```diff
@@ -33,50 +41,90 @@ class ImpersonatedServiceAccountCredentials extends CredentialsLoader implements
+     *     @type int                            $lifetime The lifetime of the impersonated credentials
+     *     @type string[]                       $delegates The delegates to impersonate
+     * }
+     * @param string|null $targetAudience The audience to request an ID token.
      */
     public function __construct(
         $scope,
-        $jsonKey
+        $jsonKey,
+        private ?string $targetAudience = null
     ) {
         if (is_string($jsonKey)) {
             if (!file_exists($jsonKey)) {
-                throw new \InvalidArgumentException('file does not exist');
+                throw new InvalidArgumentException('file does not exist');
             }
             $json = file_get_contents($jsonKey);
             if (!$jsonKey = json_decode((string) $json, true)) {
-                throw new \LogicException('invalid json for auth config');
+                throw new LogicException('invalid json for auth config');
             }
         }
         if (!array_key_exists('service_account_impersonation_url', $jsonKey)) {
-            throw new \LogicException(
+            throw new LogicException(
                 'json key is missing the service_account_impersonation_url field'
             );
         }
         if (!array_key_exists('source_credentials', $jsonKey)) {
-            throw new \LogicException('json key is missing the source_credentials field');
+            throw new LogicException('json key is missing the source_credentials field');
+        }
+        if (!array_key_exists('type', $jsonKey['source_credentials'])) {
+            throw new InvalidArgumentException('json key source credentials are missing the type field');
         }
+        if ($scope && $targetAudience) {
+            throw new InvalidArgumentException(
+                'Scope and targetAudience cannot both be supplied'
+            );
+        }
+
+        $this->targetScope = $scope ?? [];
+        $this->lifetime = $jsonKey['lifetime'] ?? 3600;
+        $this->delegates = $jsonKey['delegates'] ?? [];
 
+        $this->serviceAccountImpersonationUrl = $jsonKey['service_account_impersonation_url'];
         $this->impersonatedServiceAccountName = $this->getImpersonatedServiceAccountNameFromUrl(
-            $jsonKey['service_account_impersonation_url']
+            $this->serviceAccountImpersonationUrl
         );
 
-        $this->sourceCredentials = new UserRefreshCredentials(
-            $scope,
-            $jsonKey['source_credentials']
-        );
+        if (
+            $targetAudience !== null
+            && $jsonKey['source_credentials']['type'] === 'service_account'
+        ) {
+            // Service account tokens MUST request a scope, and as this token is only used to impersonate
+            // an ID token, the narrowest scope we can request is `cloud-platform`.
+            $scope = 'https://www.googleapis.com/auth/cloud-platform';
```

Comment:
I would like confirmation from @aeitzman that this is okay to do

Reply:
@aeitzman replied to me offline, and pointed to the python auth library, which is doing the same thing - https://github.com/googleapis/google-auth-library-python/blob/484c8db151690a4ae7b6b0ae38db0a8ede88df69/google/auth/impersonated_credentials.py#L207C1-L214C14

I've updated the scope to use the `auth/iam` scope instead, which definitely seems more appropriate.